packages/dds/map/src/directory.ts

```diff
@@ -1130,6 +1130,10 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         // Create the sub directory locally first.
         const isNew = this.createSubDirectoryCore(subdirName, true);
 
+        if (!isNew) {
+            throw new UsageError(`Directory with name ${subdirName} already exists`);
```

Comment:
I understand avoiding the unnecessary op, but why throw instead of succeeding and returning the existing subdirectory?

Reply:
Because then the user won't know that whether it created the new directory and then probably needs to do some initialization, or it already existed and may have data in it and then maybe it wants to act differently. This way it would be very clear.

Reply:
I feel ambivalent about this given the case Alex pointed out with the remote creation (and this being a breaking change). Is there a known case where a user wants this? A user could do their own checking if the distinction matters to them.